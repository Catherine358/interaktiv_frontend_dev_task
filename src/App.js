import {dataProtection, fraudPrevention, itSecurity, wphg} from './assets';

import './App.css';

function App() {
    return (
        <>
            <header>
                <div className="logo-container">
                    <p className="logo">Logo placeholder</p>
                </div>
            </header>
            <main>
                <section className="nav">
                    <div className="nav-el">
                        <div className="nav-el-subcontainer">
                            <a>Home Page</a>
                            <i className="fa-solid fa-chevron-right"></i>
                            <a>e-Learning Courses</a>
                        </div>
                    </div>
                    <div className="nav-el-admin">
                        <a>admin</a>
                    </div>
                </section>
                <section className="main">
                    <section className="sidebar-container">
                        <section className="sidebar-nav">
                            <div className="title-container">
                                <p>Navigation</p>
                            </div>
                            <div className="sidebar-nav-el-container">
                                <div className="sidebar-img-container">
                                    <img src={fraudPrevention}
                                         alt="Money Laundering and Fraud Prevention" />
                                </div>
                                <p>Money Laundering and Fraud Prevention</p>
                            </div>
                            <div className="sidebar-nav-el-container">
                                <div className="sidebar-img-container">
                                    <img src={itSecurity}
                                         alt="IT security and information security" />
                                </div>
                                <p>IT security and information security</p>
                            </div>
                            <div className="sidebar-nav-el-container">
                                <div className="sidebar-img-container">
                                    <img src={dataProtection} alt="Data Protection" />
                                </div>
                                <p>Data Protection</p>
                            </div>
                            <div className="sidebar-nav-el-container sidebar-nav-last-el">
                                <div className="sidebar-img-container">
                                    <img src={wphg} alt="WpHG-Compliance" />
                                </div>
                                <p>WpHG-Compliance</p>
                            </div>
                        </section>
                        <section className="sidebar-contact">
                            <div className="title-container">
                                <p>Contact</p>
                            </div>
                            <div className="sidebar-contact-el">
                                <p className="sidebar-contact-el-title">Technical support</p>
                                <p className="sidebar-contact-el-text">John Doe</p>
                                <a className="sidebar-contact-el-email">it@example.com</a>
                            </div>
                            <div className="sidebar-contact-el">
                                <p className="sidebar-contact-el-title">Content-didactic topics</p>
                                <p className="sidebar-contact-el-text">Jane Roe</p>
                                <a className="sidebar-contact-el-email">content@example.com</a>
                            </div>
                        </section>
                    </section>
                    <section className="courses-container">
                        <div>
                            <p className="courses-title">e-Learning Courses</p>
                        </div>
                        <div className="course-container">
                            <div className="sidebar-img-container">
                                <img src={fraudPrevention}
                                     alt="Money Laundering and Fraud Prevention" />
                            </div>
                            <div className="course-container-details">
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Intensification of money laundering prevention - private customer advice</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>30min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Basic Training - Prevention of Money Laundering and Terrorist Financing
                                            (V3.0)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>44min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Advanced Training - Prevention of Money Laundering and Terrorist Financing
                                            (V3.0)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>22min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="course-container">
                            <div className="sidebar-img-container">
                                <img src={itSecurity}
                                     alt="IT security and information security" />
                            </div>
                            <div className="course-container-details">
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>IT security basics (V1.01.3)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>40min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>March 14, 2024</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Basic training IT security and information security (V1.0)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>44min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Advanced Training IT Security and Information Security (V1.1)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>22min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="course-container">
                            <div className="sidebar-img-container">
                                <img src={dataProtection} alt="Data Protection" />
                            </div>
                            <div className="course-container-details">
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Basic training data protection (V2.0)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>0min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Privacy Basics (V1.00.2)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>40min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>In-depth training on data protection (V1.0)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>51min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>Privacy Basics 1.0</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>40min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status status-done"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                            </div>
                        </div>
                        <div className="course-container">
                            <div className="sidebar-img-container">
                                <img src={wphg} alt="WpHG-Compliance" />
                            </div>
                            <div className="course-container-details">
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>WpHG compliance basics (V1.00.3)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>30min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                                <div className="course-container-details-subcontainer">
                                    <div className="course-container-details-subcontainer-title">
                                        <p>In-depth WpHG - sales representative (V1.00.2)</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <p>30min</p>
                                        <p>To be done by:</p>
                                    </div>
                                    <div className="course-container-details-subcontainer-eval">
                                        <div className="course-container-details-subcontainer-eval-status">
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                            <span className="status"></span>
                                        </div>
                                        <p>-</p>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </section>
                </section>
            </main>
        </>
    );
}

export default App;
