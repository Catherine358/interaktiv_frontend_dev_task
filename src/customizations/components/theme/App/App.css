#main {
  background-color: #f5f5f5;
  padding-bottom: 10px;
}

body {
  /* not proud */
  min-height: 100vh !important;
  height: auto !important;
}
