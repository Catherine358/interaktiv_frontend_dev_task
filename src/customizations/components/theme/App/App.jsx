import { Provider } from 'react-redux';
import { store } from '../../../../store';
import FullView from '../FullView/FullView';
import Breadcrumbs from '../Breadcrumbs/Breadcrumbs';
import Main from '../Main/Main';

import './App.css';

function App() {
  return (
    <Provider store={store}>
      <FullView>
        <Breadcrumbs />
        <Main />
      </FullView>
    </Provider>
  );
}

export default App;
