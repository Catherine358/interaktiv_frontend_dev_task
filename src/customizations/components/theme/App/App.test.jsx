import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import App from './App';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('App', () => {
  it('renders app', () => {
    const store = mockStore({
      courses: {
        courses: [
          {
            name: 'First category',
            image: 'Some image',
            courses: [
              { title: 'First course', duration: '15min', image: 'Some image', category: 'First category', progress: 0 }
            ]
          }
        ]
      }
    });
    const component = render(
      <Provider store={store}>
        <App  />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
