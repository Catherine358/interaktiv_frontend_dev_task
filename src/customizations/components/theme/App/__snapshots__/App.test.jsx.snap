// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ui basic segment header-wrapper"
        role="banner"
      >
        <div
          class="ui container"
        >
          <div
            class="header"
          >
            <div
              class="logo"
            >
              <p>
                Logo placeholder
              </p>
            </div>
          </div>
        </div>
      </div>
      <main>
        <div
          class="ui horizontal segments"
        >
          <div
            class="ui secondary vertical segment breadcrumbs"
            role="navigation"
          >
            <div
              class="ui container"
            >
              <div
                class="ui breadcrumb"
              >
                <a
                  href="/"
                >
                  Home Page
                </a>
                <svg
                  class="icon"
                  style="height: 18px; width: auto; fill: currentColor;"
                  viewBox=""
                  xmlns=""
                />
                <div
                  class="active section"
                >
                  e-Learning Courses
                </div>
              </div>
            </div>
          </div>
          <div
            class="ui secondary vertical segment breadcrumbs admin"
            role="navigation"
          >
            <div
              class="ui container"
            >
              <div
                class="ui breadcrumb"
              >
                <div
                  class="section"
                >
                  admin
                </div>
              </div>
            </div>
          </div>
        </div>
        <section
          class="main"
        >
          <section
            class="sidebar"
          >
            <section
              class="sidebar-nav"
            >
              <div
                class="title-container"
              >
                <p>
                  Navigation
                </p>
              </div>
              <div
                class="ui segment"
                data-testid="navigation-item"
                horizontal="true"
              >
                <div
                  class="small-card-img-container"
                >
                  <img
                    alt="Money Laundering and Fraud Prevention"
                    class="ui image"
                    height="35"
                    src="fraud_prevention.png"
                    width="35"
                  />
                </div>
                <p>
                  Money Laundering and Fraud Prevention
                </p>
              </div>
              <div
                class="ui segment"
                data-testid="navigation-item"
                horizontal="true"
              >
                <div
                  class="small-card-img-container"
                >
                  <img
                    alt="IT security and information security"
                    class="ui image"
                    height="35"
                    src="it_security.png"
                    width="35"
                  />
                </div>
                <p>
                  IT security and information security
                </p>
              </div>
              <div
                class="ui segment"
                data-testid="navigation-item"
                horizontal="true"
              >
                <div
                  class="small-card-img-container"
                >
                  <img
                    alt="Data Protection"
                    class="ui image"
                    height="35"
                    src="data_protection.png"
                    width="35"
                  />
                </div>
                <p>
                  Data Protection
                </p>
              </div>
              <div
                class="ui segment"
                data-testid="navigation-item"
                horizontal="true"
              >
                <div
                  class="small-card-img-container"
                >
                  <img
                    alt="WpHG-Compliance"
                    class="ui image"
                    height="35"
                    src="wphg.png"
                    width="35"
                  />
                </div>
                <p>
                  WpHG-Compliance
                </p>
              </div>
            </section>
            <section>
              <div
                class="title-container"
              >
                <p>
                  Contact
                </p>
              </div>
              <div
                class="sidebar-contact-el"
              >
                <p
                  class="sidebar-contact-el-title"
                >
                  Technical support
                </p>
                <p
                  class="sidebar-contact-el-text"
                >
                  John Doe
                </p>
                <a
                  class="sidebar-contact-el-email"
                >
                  it@example.com
                </a>
              </div>
              <div
                class="sidebar-contact-el"
              >
                <p
                  class="sidebar-contact-el-title"
                >
                  Content-didactic topics
                </p>
                <p
                  class="sidebar-contact-el-text"
                >
                  Jane Roe
                </p>
                <a
                  class="sidebar-contact-el-email"
                >
                  content@example.com
                </a>
              </div>
            </section>
          </section>
          <section
            class="courses-container"
          >
            <div>
              <p
                class="courses-title"
                data-testid="courses-title"
              >
                e-Learning Courses
              </p>
            </div>
            <p>
              There are no courses
            </p>
          </section>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <div
      class="ui basic segment header-wrapper"
      role="banner"
    >
      <div
        class="ui container"
      >
        <div
          class="header"
        >
          <div
            class="logo"
          >
            <p>
              Logo placeholder
            </p>
          </div>
        </div>
      </div>
    </div>
    <main>
      <div
        class="ui horizontal segments"
      >
        <div
          class="ui secondary vertical segment breadcrumbs"
          role="navigation"
        >
          <div
            class="ui container"
          >
            <div
              class="ui breadcrumb"
            >
              <a
                href="/"
              >
                Home Page
              </a>
              <svg
                class="icon"
                style="height: 18px; width: auto; fill: currentColor;"
                viewBox=""
                xmlns=""
              />
              <div
                class="active section"
              >
                e-Learning Courses
              </div>
            </div>
          </div>
        </div>
        <div
          class="ui secondary vertical segment breadcrumbs admin"
          role="navigation"
        >
          <div
            class="ui container"
          >
            <div
              class="ui breadcrumb"
            >
              <div
                class="section"
              >
                admin
              </div>
            </div>
          </div>
        </div>
      </div>
      <section
        class="main"
      >
        <section
          class="sidebar"
        >
          <section
            class="sidebar-nav"
          >
            <div
              class="title-container"
            >
              <p>
                Navigation
              </p>
            </div>
            <div
              class="ui segment"
              data-testid="navigation-item"
              horizontal="true"
            >
              <div
                class="small-card-img-container"
              >
                <img
                  alt="Money Laundering and Fraud Prevention"
                  class="ui image"
                  height="35"
                  src="fraud_prevention.png"
                  width="35"
                />
              </div>
              <p>
                Money Laundering and Fraud Prevention
              </p>
            </div>
            <div
              class="ui segment"
              data-testid="navigation-item"
              horizontal="true"
            >
              <div
                class="small-card-img-container"
              >
                <img
                  alt="IT security and information security"
                  class="ui image"
                  height="35"
                  src="it_security.png"
                  width="35"
                />
              </div>
              <p>
                IT security and information security
              </p>
            </div>
            <div
              class="ui segment"
              data-testid="navigation-item"
              horizontal="true"
            >
              <div
                class="small-card-img-container"
              >
                <img
                  alt="Data Protection"
                  class="ui image"
                  height="35"
                  src="data_protection.png"
                  width="35"
                />
              </div>
              <p>
                Data Protection
              </p>
            </div>
            <div
              class="ui segment"
              data-testid="navigation-item"
              horizontal="true"
            >
              <div
                class="small-card-img-container"
              >
                <img
                  alt="WpHG-Compliance"
                  class="ui image"
                  height="35"
                  src="wphg.png"
                  width="35"
                />
              </div>
              <p>
                WpHG-Compliance
              </p>
            </div>
          </section>
          <section>
            <div
              class="title-container"
            >
              <p>
                Contact
              </p>
            </div>
            <div
              class="sidebar-contact-el"
            >
              <p
                class="sidebar-contact-el-title"
              >
                Technical support
              </p>
              <p
                class="sidebar-contact-el-text"
              >
                John Doe
              </p>
              <a
                class="sidebar-contact-el-email"
              >
                it@example.com
              </a>
            </div>
            <div
              class="sidebar-contact-el"
            >
              <p
                class="sidebar-contact-el-title"
              >
                Content-didactic topics
              </p>
              <p
                class="sidebar-contact-el-text"
              >
                Jane Roe
              </p>
              <a
                class="sidebar-contact-el-email"
              >
                content@example.com
              </a>
            </div>
          </section>
        </section>
        <section
          class="courses-container"
        >
          <div>
            <p
              class="courses-title"
              data-testid="courses-title"
            >
              e-Learning Courses
            </p>
          </div>
          <p>
            There are no courses
          </p>
        </section>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
